### ISSUE TITLE
BesuCommand.isRevertReasonEnabled belongs in DataStorageConfiguration

### ISSUE THREAD

**Issue by jflo** [hyperledger/besu#7597]
While doing some refactoring I had to provide `isRevertReasonEnabled` from BesuCommand through the BesuController, and it struck me that this is more obviously an EVM configuration option. This should be refactored and folded into EVMConfiguration.

**Comment by shemnon** [comment]
Isn't this a data storage question?  Revert Reason is in the output of the root message frame, wether or not it is recorded.  No code in the EVM module uses it, the flag is part of `TransactionReceiptFactory`, switching between these two functions - https://github.com/hyperledger/besu/blob/8eee5698878af80a543362d136a01c51099d48d6/ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/MainnetProtocolSpecs.java#L940-L958.

When writing it to trie you already have the flag https://github.com/hyperledger/besu/blob/15d54afdf9324ed66d6c6cbad493b18fcb318714/ethereum/core/src/main/java/org/hyperledger/besu/ethereum/core/TransactionReceipt.java#L191, so we could simply drop it from the protocol spec factory instead.

**Comment by siladu** [comment]
@jflo is this blocking anything else? Should we prioritise on a team's board or happy to leave for anyone to pick up?

**Comment by jflo** [comment]
Anyone can pick this up, and should consider @shemnon 's suggestions above. I'm not married to any particular approach, as long as this option ends up being grouped in somewhere more specific and relevant.

**Comment by bomanaps** [comment]
Hello @jflo  & @shemnon ,

I've reviewed this issue and would like to confirm my understanding of the steps required to address it:

1. Move isRevertReasonEnabled to EVMConfiguration:
   - Add a new field isRevertReasonEnabled in the EVMConfiguration class
   - Implement getter and setter methods for this field

2. Update BesuCommand and BesuController:
   - Remove direct handling of isRevertReasonEnabled in BesuCommand
   - Modify BesuController to initialize and manage this flag using EVMConfiguration

3. Modify TransactionReceiptFactory:
   - Update methods to use EVMConfiguration for checking the isRevertReasonEnabled flag
   - Combine the two factory methods into one, where the revert reason is included or excluded based on EVMConfiguration

4. Update MainnetProtocolSpecs:
   - Change logic to pass EVMConfiguration to the TransactionReceiptFactory when creating transaction receipts

If these steps accurately reflect the requirements, I would like to request assignment to this issue. Thank you for your consideration.

**Comment by shemnon** [comment]
@bomanaps Do not, under any condition, add it to `EVMConfiguration`.  Do some more studying of where and how the flag is used.  EVMConfiguration should only be for options used by the `:evm` module, and it isn't used there.  In fact, you will discover that the flag is only ever used in the `ProtocolSpec`

Adding it to `DataStorageConfiguration` would be a better place.  Then `MainnetProtocolSpecs` and `ClassicProtocolSpecs` can replace the two fields `enableRevertReason` and `isParallelTxProcessingEnabled` parameters being passed around with the single `DataStorageConfiguration` Object.


We could move `chainId` into `EVMConfiguration`, that refactoring would make sense.  Probably in a different PR as we would also want to update `ChainIdOperation` as well as any other place ChainId and EVMConfiguration are passed along together.  But that seems like a separate PR

**Comment by bomanaps** [comment]
Sorry for the misunderstanding. I will move isRevertReasonEnabled to DataStorageConfiguration, include this flag (along with isParallelTxProcessingEnabled), and remove it from BesuCommand and BesuController. I'll also update MainnetProtocolSpecs and ClassicProtocolSpecs accordingly.

**Comment by jflo** [comment]
yes @bomanaps , you've proposed a better design here, feel free to proceed.

**Comment by macfarla** [comment]
@bomanaps any progress on this one?

**Comment by bomanaps** [comment]
> [@bomanaps](https://github.com/bomanaps) any progress on this one?

Hello, I have tried countless times to build but keep it this issue , plus build process before the error is usually an hr plus

<img width="960" height="584" alt="Image" src="https://github.com/user-attachments/assets/99106c9d-9442-4b84-befd-2ce6c5db3276" />

**Comment by macfarla** [comment]
@bomanaps suggestions
* does the code compile - `./gradlew compileTestJava`
* can you run the tests for just the files you've changed (and those ones that are failing) to catch likely failures, rather than running the full `./gradlew build` which yes does take some time - you can run individual tests via your IDE or on the command line eg `./gradlew :ethereum:api:test --tests org.hyperledger.besu.ethereum.api.jsonrpc.internal.methods.engine.EngineForkchoiceUpdatedV2Test`
* also suggest looking at what the failing tests are doing, how do they behave before your changes vs after
* once you've tried that, if you're still stuck, push a PR so we can see your code changes and can go from there.

**Comment by bomanaps** [comment]
> [@bomanaps](https://github.com/bomanaps) suggestions
> 
> * does the code compile - `./gradlew compileTestJava`
> * can you run the tests for just the files you've changed (and those ones that are failing) to catch likely failures, rather than running the full `./gradlew build` which yes does take some time - you can run individual tests via your IDE or on the command line eg `./gradlew :ethereum:api:test --tests org.hyperledger.besu.ethereum.api.jsonrpc.internal.methods.engine.EngineForkchoiceUpdatedV2Test`
> * also suggest looking at what the failing tests are doing, how do they behave before your changes vs after
> * once you've tried that, if you're still stuck, push a PR so we can see your code changes and can go from there.

I have not made the implementation, am just trying to run build first which I usually do, and am getting stuck here, if I can’t build it with no code changes am not sure how am going to run the code after I have made the implementation to check what I have implemented

**Comment by macfarla** [comment]
have you checked out latest main branch? What version of java are you using? Besu requires Java 21. ClassCastException is not expected. Does this command work `./gradlew compileTestJava`

**Comment by Apisapple** [comment]
Hi @jflo, I’m interested in contributing to Besu and would like to work on this issue.

I noticed that this issue is currently assigned to @bomanaps, but there hasn’t been recent activity.  
Is this issue still being worked on? If it is already in progress, could you recommend another good first issue with similar difficulty?

Thank you!

**Comment by abhigyan1102** [comment]
Hi @jflo, I see kkaur01 started work on this (PR #9858) but appears to have stepped away. Is this still available to pick up? I'd be happy to either continue from their work or start fresh with a clean implementation following @shemnon's guidance — moving isRevertReasonEnabled into DataStorageConfiguration and consolidating it with isParallelTxProcessingEnabled.

**Comment by Utkal059** [comment]
Hi @jflo , since this is currently unassigned and there hasn't been a response to the recent interest, I'd like to go ahead and pick this up. I will follow @shemnon's guidance—moving isRevertReasonEnabled into DataStorageConfiguration alongside isParallelTxProcessingEnabled. I wll open a PR shortly.

**Comment by abhay-dev2901** [comment]
Hi @jflo, since this issue is currently unassigned and there has not been a maintainer response to the recent interest, I’d like to
  pick this up.
  I’ll follow @shemnon’s guidance by moving `isRevertReasonEnabled` into `DataStorageConfiguration` and removing the separate plumbing
  through `BesuCommand`/`BesuControllerBuilder`. I’ll keep the change focused and open a PR shortly.


  Please let me know if someone is already actively working on this.

**Comment by abhay-dev2901** [comment]
@jflo @shemnon Linked the PR here for your reference.
